source/fe/fe_rt_bubbles.cc

```diff
@@ -0,0 +1,288 @@
+
+
+template <int dim>
+FE_RT_Bubbles<dim>::FE_RT_Bubbles (const unsigned int deg)
+  :
+  FE_PolyTensor<PolynomialsRT_Bubbles<dim>, dim> (
+    deg,
+    FiniteElementData<dim>(get_dpo_vector(deg),
+                           dim, deg+1, FiniteElementData<dim>::Hdiv),
+    get_ria_vector (deg),
+    std::vector<ComponentMask>(PolynomialsRT_Bubbles<dim>::compute_n_pols(deg),
+                               std::vector<bool>(dim,true)))
+{
+  Assert (dim >= 2, ExcImpossibleInDim(dim));
+  Assert (deg >= 1, ExcMessage("Lowest order RT_Bubbles element is degree 1, but you requested for degree 0"));
+  const unsigned int n_dofs = this->dofs_per_cell;
+
+  this->mapping_type = mapping_raviart_thomas;
+  // Initialize support points and quadrature weights
+  initialize_support_points(deg);
+  // Compute the inverse node matrix to get
+  // the correct basis functions
+  FullMatrix<double> M = FETools::compute_node_matrix(*this);
+  this->inverse_node_matrix.reinit(n_dofs, n_dofs);
+  this->inverse_node_matrix.invert(M);
+
+  // Reinit the vectors of prolongation matrices to the
+  // right sizes. There are no restriction matrices implemented
+  for (unsigned int ref_case=RefinementCase<dim>::cut_x;
+       ref_case<RefinementCase<dim>::isotropic_refinement+1; ++ref_case)
+    {
+      const unsigned int nc = GeometryInfo<dim>::n_children(RefinementCase<dim>(ref_case));
+
+      for (unsigned int i=0; i<nc; ++i)
+        this->prolongation[ref_case-1][i].reinit (n_dofs, n_dofs);
+    }
+  // Fill prolongation matrices with embedding operators
+  // set tolerance to 1, as embedding error accumulate quickly
+  FETools::compute_embedding_matrices (*this, this->prolongation, true, 1.0);
+  FullMatrix<double> face_embeddings[GeometryInfo<dim>::max_children_per_face];
+  for (unsigned int i=0; i<GeometryInfo<dim>::max_children_per_face; ++i)
+    face_embeddings[i].reinit (this->dofs_per_face, this->dofs_per_face);
+  FETools::compute_face_embedding_matrices<dim,double>(*this, face_embeddings, 0, 0);
+  this->interface_constraints.reinit((1<<(dim-1)) * this->dofs_per_face,
+                                     this->dofs_per_face);
+  unsigned int target_row=0;
+  for (unsigned int d=0; d<GeometryInfo<dim>::max_children_per_face; ++d)
+    for (unsigned int i=0; i<face_embeddings[d].m(); ++i)
+      {
+        for (unsigned int j=0; j<face_embeddings[d].n(); ++j)
+          this->interface_constraints(target_row,j) = face_embeddings[d](i,j);
+        ++target_row;
+      }
+}
+
+
+
+template <int dim>
+std::string
+FE_RT_Bubbles<dim>::get_name () const
+{
+  // Note: this->degree is the maximal polynomial degree and is thus one higher
+  // than the argument given to the constructor
+  std::ostringstream namebuf;
+  namebuf << "FE_RT_Bubbles<" << dim << ">(" << this->degree << ")";
+
+  return namebuf.str();
+}
+
+
+
+template <int dim>
+std::unique_ptr<FiniteElement<dim,dim>>
+                                     FE_RT_Bubbles<dim>::clone() const
```

Comment:
you can fix the odd indentation if you put a space between `> >` in line 106 -- the astyle version we use hasn't learned about C++11 yet